python/python/bystro/proteomics/proteomics_cli.py

```diff
@@ -0,0 +1,158 @@
+"""Provide a CLI for proteomics analysis."""
+import argparse
+from pathlib import Path
+from typing import Any, BinaryIO
+
+import requests
+from bystro.api.cli import DEFAULT_DIR, authenticate, login
+from msgspec import json as mjson
+
+# ruff: noqa: T201
+
+UPLOAD_PROTEIN_ENDPOINT = "/api/jobs/upload_protein/"
```

Comment:
Right now we organize connect all resources to a single "job" and job_id. A job is a collection of resources, minimally an annotation (+ ancestry, search index, etc).
  - behind the scenes, each new resource is a sub-document on the main annotation document, see https://mongoosejs.com/docs/subdocs.html

I think a straightforward next step is to add a new kind of resource, protein abundance, as another subdocument in job.

We could alternatively de-couple the two.

Which would you prefer, and why?

Reply:
>  I had assumed that jobs were more like descriptions of computations to be performed, but if they're just collections of resources that clears up some questions for me.

Our conceptions are equivalent if you consider a promise a resource

Reply:
> We allow creating a new job with only proteomics data (no annotation). This will require some updating to the frontend, to ensure that the user can add an annotation to an existing job.

That makes sense to me. Long term if we want to be a general life sciences inference platform, we probably need to be able to support resource separately.

To keep things simple, in iteration one, I'll generalize the concept of job, such that a job can be created without an annotation, and that annotation is just another promise for a resource.